scheduler/scheduler.go

```diff
@@ -139,22 +146,87 @@ func spawnHosts(ctx context.Context, newHostsNeeded map[string]int) (map[string]
 			continue
 		}
 
+		if ctx.Err() != nil {
+			return nil, errors.New("scheduling run canceled.")
+		}
+
+		d, err := distro.FindOne(distro.ById(distroId))
+		if err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
```

Comment:
This should return the error instead of logging

Reply:
+1